### ISSUE TITLE
Add `@maintenance` decorator to methods that cannot be executed concurrently

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/disk-objectstore#6]
Add a guard to avoid maintenance methods (repack, delete, add directly to packs) are performed in parallel.

Add decorator that opens a lock file in exclusive mode, and stops if someone else is already performing maintenance operations.

Possibly check (both this, and the one that locks a pack) that they have a finally block that releases the lock.

**Comment by giovannipizzi** [comment]
Also:
- check if there is a second level of locking, e.g. when deleting?
- until #4 is fixed, check if we need to prevent also normal usage of the repository by any process while in maintenance? (E.g. with a lock file, containing the PID, and in case of failure of the packing process, we explain how to check if it's still running, and how to delete the lock).

**Comment by giovannipizzi** [comment]
In AiiDA, we will try to put all safeguards to avoid problems. Putting a decorator properly is a bit tricky, so I am moving this to a 'nice to have'. This will need to be properly documented, though...
